### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by mdhaber** [mdhaber/marray#15]
I think functions/methods/attributes should mirror the documentation of the underlying array library. In addition, though, we can include a standard note about masks and link to a tutorial that covers how masks affect the calculation in each type of function.

**Comment by keewis** [comment]
I don't think documentation is going to be easy, as the namespace (and the class and all the functions within it) are constructed dynamically.

**Comment by mdhaber** [comment]
> I don't think documentation is going to be easy, as the namespace

And all I mean to do initially is to generate documentation dynamically: copy the underlying array documentation, and add a note about the mask behavior corresponding with the type of function it is. For instance, all binary elementwise operations will get the same standard note about how the mask of the result is the OR of the masks of the two arguments.

The other part is a tutorial, which shows basic use and each of the distinct mask behaviors. 

We've recently been able to add in-browser executable notebook tutorials to SciPy that we can write in markdown. If you know or are interested in how to do that sort of thing, one thing that would really help (like setting up CI was! Thank you!) would be to get a placeholder in for the tutorial. If worst comes to worst, though, I'll just link from the GitHub landing page to a Colab notebook.

**Comment by mdhaber** [comment]
I improved the readme and added a draft tutorial [here](https://colab.research.google.com/drive/1LaZCK3jvnf40qEjWEhqhhc5e8IWy7vuo?usp=sharing). Also, as of gh-37:
```shell
>>> help(mxp.asarray)
Help on function asarray in module marray:
asarray(obj, /, *, mask=None, dtype=None, device=None, copy=None)
    The following is the documentation for the corresponding attribute of `numpy`.
    The behavior of `marray` is the same when the mask is all `False`, and differs (as
    described in the tutorial) when the mask has `True` elements.
    asarray(a, dtype=None, order=None, *, device=None, copy=None, like=None)

        Convert the input to an array.
...
```

To close this issue, the only thing I might like to do is to show the tutorial on a website of our own, if that's easy. I don't really know anything about this sort of thing, so @keewis, LMK if this is something you can do. If not, I'll probably just add the notebook to a `docs` folder and link to it with [nbviewer](https://nbviewer.org/) rather than with my personal Colab account.

Improvements to the tutorial and `help`-accessible documentation are separate matters; the point is that we have basic documentation.

**Comment by mdhaber** [comment]
I set up basic documentation with Jupyter Books published on GitHub Pages [here](https://mdhaber.github.io/marray/intro.html). Now, the tutorial is a markdown notebook, so it will be easier to edit. I would like the result cells to render a bit differently from the main code cells, and it would be awesome if we could make this executable in browser, but I think it's enough to say that we've added documentation. PRs to improve it welcome.
